File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        int cnt = 0;
        ListNode * dummy = new ListNode(0);
        dummy->next = head;
        ListNode * tmp = dummy;
        while(tmp){
            tmp = tmp->next;
            cnt++;
        }
        ListNode * prev = NULL;
        tmp = dummy;
        for(int i=0;i<cnt-n;i++){
            prev = tmp;
            tmp = tmp->next;
        }
        prev->next = tmp->next;
        tmp->next = NULL;
        delete tmp;
        return dummy->next;
        
    }
};

File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        if not head:
            return head
        sizecnt = 0
        tmp = head
        while(tmp):
            sizecnt = sizecnt+1
            tmp = tmp.next
        if(sizecnt==1):
            return None
        if sizecnt==n:
            return head.next
        tmp = head
        for i in range(sizecnt-n-1):
            tmp = tmp.next
            
        # print(tmp.val)
        tmp.next = tmp.next.next
        tmp = tmp.next
        del tmp
        return head
        
            